### PR TITLE
tiproxy: add vip configurations

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

1. Add virtual IP-related doc.
2. Remove Prometheus-related limitations in load balance. If Prometheus is unavailable, TiProxy will query TiDB directly.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18079
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by djshow832 — APPROVED** [review]

File: tiproxy/tiproxy-configuration.md [hunk 1]
```diff
@@ -118,6 +119,28 @@ Configurations for the load balancing policy of TiProxy.
...
 
+### ha
+
+High availability configurations for TiProxy.
+
+#### `virtual-ip`
+
++ Default value: ``
++ Support hot-reload: no
++ Specifies the virtual IP address in CIDR format, such as `10.0.1.10/24`. In a cluster with multiple TiProxy instances, only one instance binds to the virtual IP. If this instance goes offline, another TiProxy instance will take over the IP, ensuring clients can always connect to an available TiProxy through the virtual IP.
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:130]:
```suggestion
+ Specifies the virtual IP address in CIDR format, such as `10.0.1.10/24`. In a cluster with multiple TiProxy instances, only one instance binds to the virtual IP. If this instance goes offline, another TiProxy instance will automatically bind to the IP, ensuring clients can always connect to an available TiProxy through the virtual IP.
```

File: tiproxy/tiproxy-configuration.md [hunk 2]
```diff
@@ -118,6 +119,28 @@ Configurations for the load balancing policy of TiProxy.
...
 
+### ha
+
+High availability configurations for TiProxy.
+
+#### `virtual-ip`
+
++ Default value: ``
++ Support hot-reload: no
++ Specifies the virtual IP address in CIDR format, such as `10.0.1.10/24`. In a cluster with multiple TiProxy instances, only one instance binds to the virtual IP. If this instance goes offline, another TiProxy instance will automatically bind to the IP, ensuring clients can always connect to an available TiProxy through the virtual IP.
```
Comment (qiancai) [inline tiproxy/tiproxy-configuration.md:130]:
```suggestion
+ Specifies the virtual IP address in the CIDR format, such as `10.0.1.10/24`. In a cluster with multiple TiProxy instances, only one instance binds to the virtual IP. If this instance goes offline, another TiProxy instance will automatically bind to the IP, ensuring clients can always connect to an available TiProxy through the virtual IP.
```

File: tiproxy/tiproxy-overview.md [hunk 3]
```diff
@@ -91,7 +91,7 @@ This section describes how to deploy and change TiProxy using TiUP. For how to d
 
 3. Configure the TiProxy instances.
 
-    To ensure the high availability of TiProxy, it is recommended to deploy at least two TiProxy instances. You can use hardware load balancers to distribute traffic to each TiProxy instance, or configure virtual IP to route the traffic to the available TiProxy instance.
+    To ensure the high availability of TiProxy, it is recommended to deploy at least two TiProxy instances and configure the virtual IP by setting [`ha.virtual-ip`](/tiproxy/tiproxy-configuration.md#virtual-ip) and [`ha.interface`](/tiproxy/tiproxy-configuration.md#interface) to route the traffic to the available TiProxy instance.
```
Comment (qiancai) [inline tiproxy/tiproxy-overview.md:94]:
```suggestion
    To ensure the high availability of TiProxy, it is recommended to deploy at least two TiProxy instances and configure a virtual IP by setting [`ha.virtual-ip`](/tiproxy/tiproxy-configuration.md#virtual-ip) and [`ha.interface`](/tiproxy/tiproxy-configuration.md#interface) to route the traffic to the available TiProxy instance.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: tiproxy/tiproxy-configuration.md [hunk 4]
```diff
@@ -118,6 +119,28 @@ Configurations for the load balancing policy of TiProxy.
 + Possible values: `resource`, `location`, `connection`
 + Specifies the load balancing policy. For the meaning of each possible value, see [TiProxy load balancing policies](/tiproxy/tiproxy-load-balance.md#configure-load-balancing-policies).
 
+### ha
+
+High availability configurations for TiProxy.
+
+#### `virtual-ip`
+
++ Default value: ``
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:128]:
```suggestion
+ Default value: `""`
```

File: tiproxy/tiproxy-configuration.md [hunk 5]
```diff
@@ -118,6 +119,28 @@ Configurations for the load balancing policy of TiProxy.
...
+
+> **Note:**
+>
+> - Virtual IP is only supported on Linux operating systems.
+> - The Linux user running TiProxy must have permission to bind IP addresses.
+> - The virtual IP and the IPs of all TiProxy instances must be within the same CIDR range.
+
+#### `interface`
+
++ Default value: ``
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:140]:
```suggestion
+ Default value: `""`
```

File: tiproxy/tiproxy-configuration.md [hunk 6]
```diff
@@ -118,6 +119,28 @@ Configurations for the load balancing policy of TiProxy.
...
+>
+> - Virtual IP is only supported on Linux operating systems.
+> - The Linux user running TiProxy must have permission to bind IP addresses.
+> - The virtual IP and the IPs of all TiProxy instances must be within the same CIDR range.
+
+#### `interface`
+
++ Default value: `""`
++ Support hot-reload: no
++ Specifies the network interface to bind the virtual IP to, such as `eth0`. The virtual IP will be bound to a TiProxy instance only when both [`ha.virtual-ip`](#virtual-ip) and `ha.interface` are set.
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:142]:
```suggestion
+ Specifies the network interface to bind the virtual IP to, such as `"eth0"`. The virtual IP will be bound to a TiProxy instance only when both [`ha.virtual-ip`](#virtual-ip) and `ha.interface` are set.
```

**Pull request merged**